src/id/mangacan/src/eu/kanade/tachiyomi/extension/id/mangacan/MangaCan.kt

```diff
@@ -0,0 +1,118 @@
+package eu.kanade.tachiyomi.extension.id.mangacan
+
+import eu.kanade.tachiyomi.multisrc.mangathemesia.MangaThemesia
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import rx.Observable
+
+class MangaCan : MangaThemesia(
+    "Manga Can",
+    "https://mangacanblog.com",
+    "id",
+    "/",
+) {
+    override val client = super.client.newBuilder()
+        .rateLimit(3)
+        .build()
+
+    override val supportsLatest = false
+
+    override val seriesGenreSelector = ".seriestugenre a[href*=genre]"
+
+    override val pageSelector = "div.images img"
+
+    private var genreList: Array<Pair<String, String>> = emptyArray()
+
+    override fun imageRequest(page: Page): Request {
+        return super.imageRequest(page).newBuilder()
+            .removeHeader("Referer")
+            .addHeader("Referer", "$baseUrl/")
+            .build()
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        if (genreList.isEmpty()) {
+            genreList += "All" to ""
+            genreList += parseGenres(response.asJsoup(response.peekBody(Long.MAX_VALUE).string()))
+        }
+
+        return super.searchMangaParse(response)
+    }
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        if (query.startsWith(URL_SEARCH_PREFIX).not()) return super.fetchSearchManga(page, query, filters)
+        val mangaPath = query.substringAfterLast(baseUrl)
+        return fetchMangaDetails(SManga.create().apply { this.url = mangaPath })
+            .map {
+                it.url = mangaPath
```

Comment:
Not too familiar with the `fetch*` methods, but is there a reason you can't use `setUrlWithoutDomain()`?